### PR TITLE
Refactor and add tests for `pathgetlink()` function

### PR DESCRIPTION
**Pull request by siteshwar**

### CODE REVIEW

File: src/lib/libast/tests/path/pathgetlink.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include "ast.h"
+#include "terror.h"
+
+tmain() {
+    UNUSED(argc);
+    UNUSED(argv);
+    const char *linkname = "foo";
+
+    char buff[10];
+    char filename[] = "foobarbaz";
```
Comment (krader1961) [inline src/lib/libast/tests/path/pathgetlink.c]: Why isn't this `const char *filename = "foobarbaz";`?

File: src/lib/libast/path/pathgetlink.c [hunk 2]
```diff
@@ -31,19 +31,19 @@
...
-/*
- * return external representation for symbolic link text of name in buf
- * the link text string length is returned
- */
-
-int pathgetlink(const char *name, char *buf, int siz) {
+//
+// Return external representation for symbolic link text of name in buff
+// the link text string length is returned
+///
```
Comment (krader1961) [inline src/lib/libast/path/pathgetlink.c]: Just two forward slashes will do :smile:

**Comment by krader1961** [comment]
LGTM modulo two minor nits.

File: src/lib/libast/tests/path/pathgetlink.c (hunk 1 above)
Comment (siteshwar) [reply src/lib/libast/tests/path/pathgetlink.c]: I was playing around with some other functions that required this variable to be writable, but will fix it.

**Comment by krader1961** [comment]
FWIW, There is far too little use of the `const` qualifier in this project where using that qualifier would help to identify bugs or APIs that are ill-defined.

**Pull request merged**